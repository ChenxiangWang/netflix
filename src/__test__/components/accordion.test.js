import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Accordion from "../../components/accordion";
describe("<Accordion>", () => {
  test("render <Accordion>", () => {
    const { container } = render(
      <Accordion>
        <Accordion.Title>title</Accordion.Title>
        <Accordion.Item>
          <Accordion.Header data-testid="clickable">header</Accordion.Header>
          <Accordion.Body>body</Accordion.Body>
        </Accordion.Item>
      </Accordion>
    );

    expect(screen.getByText("title")).toBeTruthy();
    expect(screen.getByText("header")).toBeTruthy();
    expect(screen.queryByText("body")).toBeFalsy();

    fireEvent.click(screen.queryByTestId("clickable"));
    expect(screen.getByText("body")).toBeTruthy();

    fireEvent.click(screen.queryByTestId("clickable"));
    expect(screen.queryByText("body")).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
