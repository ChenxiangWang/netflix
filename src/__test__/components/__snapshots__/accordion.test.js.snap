// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion> render <Accordion> 1`] = `
<div
  class="style__Container-sc-188x0cs-0 fIUgdX"
>
  <div
    class="style__Inner-sc-188x0cs-1 kjINHE"
  >
    <h1
      class="style__Title-sc-188x0cs-2 gOOTZl"
    >
      title
    </h1>
    <div
      class="style__Item-sc-188x0cs-3 GjnSX"
    >
      <div
        class="style__Header-sc-188x0cs-4 jdrWa-D"
        data-testid="clickable"
      >
        header
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
